code/graphics/software/font.cpp

```diff
@@ -250,7 +250,10 @@ namespace
 			}
 
 			for (auto i = 0; i < (int)Lcl_languages.size(); ++i) {
-				Lcl_languages[i].special_char_indexes[font_id] = (ubyte)default_special_char_index;
+				while (font_id > (int)Lcl_languages[i].special_char_indexes.size()) {
+					Lcl_languages[i].special_char_indexes.push_back(0);
+				}
+				Lcl_languages[i].special_char_indexes.push_back((ubyte)default_special_char_index);
```

Comment:
Is the `push_back` actually correct here?  It seems to be that you still need to access the `[font_id]` entry.

Reply:
I think here I was working under the assumption that the font specified would be a new font, but if was already specified, I guess we do need to access through [font_id].  I have made that change in the last commit.

Reply:
Yeah, I'm referring to the second `push_back` here, not the first one.

Reply:
Right, in the event that that item in the vector doesn't exist yet, it creates any default entries that need to exist in between and then the second `push_back()` places it at the font_id index.